src/participant.ts

```diff
@@ -309,14 +324,28 @@ export class Participant extends EventEmitter {
   private attachListeners() {
     this.rpc = new RPC(this.frame.contentWindow, '1.0');
 
-    this.rpc.expose('sendInteractivePacket', data => {
+    this.rpc.expose<{ method: string; params: any }>('sendInteractivePacket', data => {
       this.websocket!.send(
         JSON.stringify({
           ...data,
           type: 'method',
           discard: true,
         }),
       );
+
+      if (data.method !== 'giveInput') {
+        return;
+      }
+
+      const kind = this.controls.getControlKind(data.params.controlID);
+      if (!kind) {
+        return;
+      }
+
+      this.emit('input', {
+        ...data.params,
+        kind,
```

Comment:
Instead it would be great if we can get a small middleware model going. For instance, one thing we'll have with certain changes is a prompt to log in when we get an Interactive packet from the server. Being able to intercept that would be useful. Right now this same thing can be accomplished by listening to `recvMethod` on the Participant's RPC.